wts.go

```diff
@@ -0,0 +1,435 @@
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSInitialProgram)
+}
+
+func (wts *WTS) QuerySessionApplicationName(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSApplicationName)
+}
+func (wts *WTS) QuerySessionWorkingDirectory(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSWorkingDirectory)
+}
+
+func (wts *WTS) QuerySessionID(sessionID uint32) (uint32, error) {
+	return wts.querySessionInformationAsUint32(sessionID, wrappers.WTSSessionId)
+}
+
+func (wts *WTS) QuerySessionUserName(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSUserName)
+}
+
+func (wts *WTS) QuerySessionWinStationName(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSWinStationName)
+}
+
+func (wts *WTS) QuerySessionDomainName(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSDomainName)
+}
+
+func (wts *WTS) QuerySessionConnectState(sessionID uint32) (ConnectState, error) {
+	r1, err := wts.querySessionInformationAsUint32(sessionID, wrappers.WTSConnectState)
+	return ConnectState(r1), err
+}
+
+func (wts *WTS) QuerySessionClientBuildNumber(sessionID uint32) (uint32, error) {
+	return wts.querySessionInformationAsUint32(sessionID, wrappers.WTSClientBuildNumber)
+}
+
+func (wts *WTS) QuerySessionClientName(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSClientName)
+}
+
+func (wts *WTS) QuerySessionClientDirectory(sessionID uint32) (string, error) {
+	return wts.querySessionInformationAsString(sessionID, wrappers.WTSClientDirectory)
+}
+
+func (wts *WTS) QuerySessionClientProductId(sessionID uint32) (uint16, error) {
+	return wts.querySessionInformationAsUint16(sessionID, wrappers.WTSClientProductId)
+}
+
+func (wts *WTS) QuerySessionClientHardwareId(sessionID uint32) (uint32, error) {
+	return wts.querySessionInformationAsUint32(sessionID, wrappers.WTSClientHardwareId)
+}
+
+func (wts *WTS) QuerySessionClientAddress(sessionID uint32) (net.IP, error) {
+	var buffer uintptr
+	var bytesReturned uint32
+
+	if err := wrappers.WTSQuerySessionInformation(wts.handle, sessionID, wrappers.WTSClientAddress, &buffer, &bytesReturned); err != nil {
+		return net.IP{}, err
+	}
+	defer wrappers.WTSFreeMemory(unsafe.Pointer(buffer))
+
+	if bytesReturned != 24 {
+		return net.IP{}, buferSizeError(24, bytesReturned)
+	}
+	// MS doc: The IP address is offset by two bytes from the start of the Address member of the WTS_CLIENT_ADDRESS structure.
+	// https://msdn.microsoft.com/en-us/library/aa383861%28v=vs.85%29.aspx
+	a := *(*wrappers.WTS_CLIENT_ADDRESS)(unsafe.Pointer(buffer))
+	return clientAddressToIP(a.AddressFamily, a.Address[2:])
+}
+
+func (wts *WTS) QuerySessionClientDisplay(sessionID uint32) (wrappers.WTS_CLIENT_DISPLAY, error) {
```

Comment:
We should define a Go-style structure and return that in place of `WTS_CLIENT_DISPLAY`.